### DISCUSSION TITLE
Questions for input data download

### DISCUSSION THREAD

**Discussion by ZhangBomei** [WAM2layers/WAM2layers#285] [Q&A]
Hello! I want to use the WAM2layers model for long-term water vapor tracking. However, the input data for multiple years is very large, measured in terabytes (TB). The data download speed is also slow. I'm wondering if it's possible to process the input data using cloud computing without the necessity of downloading. Additionally, are there any other methods to download the data faster?

**Comment by Peter9192** [comment]
Hi @ZhangBomei. Interesting question! Are you planning to use ERA5? I think there are copies of ERA5 in cloud storage at some places, but I've never worked with those before. I would think it should be possible to do what you suggest. A few points

- An analysis with WAM2layers is a 2-step process. The first step is pre-processing, so you would need to store the output of this somewhere/temporarily/in the cloud. Then you can do the actual tracking. Preprocessed data is smaller than original, as the data is reduced to 2 layers.
- Currently the example preprocessing script for era5 assumes data comes in netcdf format. Cloud storage could be different. 
- If you have access to data that is in cloud-optimized storage such as zarr, you might be able to modify the preprocessing script such that it reads in the remote data directly as an alternative way of downloading the data.

**Reply by ZhangBomei** [reply]
Yes! I want to use ERA5. When I sought another way to download  ERA5 on the website, some people mentioned Google Earth Engine. Perhaps it is possible to use Google Earth Engine for importing input data, at the pre-processing step?

**Reply by XINGXING-ZHANG** [reply]
@ZhangBomei I don't think data downloading should be a big problem. You can narrow down the data range according to your needs, and it doesn't necessarily require a global scope. In addition, I have also downloaded ERA data with high global resolution (0.1 degree), which inevitably takes some time to download. I just want you to know that I have had similar experiences, and I hope it will be helpful to you.

**Reply by Peter9192** [reply]
I'm not very familiar with google earth engine, but if I understand correctly that would mean you do the calculations inside GEE, i.e. on their resources, right? I guess you could do that and then download the pre-processed data, or perhaps also perform the tracking inside GEE. You might have to figure out whether you can do this free of charge, or whether you need to pay for the compute. I'd be interested to learn about your experience 😄

**Reply by ZhangBomei** [reply]
@XINGXING-ZHANG Thank you for sharing your experience! As you say, waiting for download data is unavoidable. In fact, I’ve been trying to narrow down my experimental area to reduce the size of the downloaded data.

**Reply by ZhangBomei** [reply]
@Peter9192  Thank you for your response! Because I know GEE could process remote sensing data and output the processed results directly. So, when I knew that perhaps GEE could be used to download data, I thought it was possible to use GEE for importing input data at the pre-processing step and outputting the pre-processed result directly. I’m sorry, it seems that GEE doesn't provide direct access to ERA5 hourly data (0.25 degree). And how to use GEE to achieve the pre-processing step is also a big challenge,it is a complicated project. But if I have any other progress, I will share my experience here. Thank you for your reply again!
